tests/test_multiaddr.py

```diff
@@ -246,21 +227,6 @@ def test_bad_initialization_wrong_type():
         Multiaddr(42)
 
 
-def test_get_value_too_many_fields_protocol(monkeypatch):
-    """
-    This test patches the Multiaddr's string representation to return
-    an invalid string in order to test that value_for_protocol properly
-    throws a ValueError.  This avoids some of the error checking in
-    the constructor and is easier to patch, thus the actual values
-    that the constructor specifies is ignored by the test.
-    """
-    monkeypatch.setattr("multiaddr.multiaddr.Multiaddr.__str__",
-                        lambda ignore: str('/udp/1234/5678'))
-    a = Multiaddr("/ip4/127.0.0.1/udp/1234")
-    with pytest.raises(ValueError):
-        a.value_for_protocol(P_UDP)
```

Comment:
Why is this test removed?

Reply:
`value_for_protocol` does not parse the multiaddr's string representation anymore, but rather just uses the binary representation already present in the `Multiaddr` object. This avoids a completely unnecessary two-way conversion we previously had, but it means that this test doesn't test anything anymore since the return value of `__str__` is moot.